src/packages/DAO/elections.dao.ts

```diff
@@ -38,31 +44,55 @@ export async function createElection(formData: CreateElection) {
 export async function startElection(formData: FormData) {
   const electionId = String(formData.get("electionId"));
   const updateElectionSchema = z.string().uuid();
-  const parsedElectionId = updateElectionSchema.parse(electionId);
+  const parsedElectionId = updateElectionSchema.safeParse(electionId);
+
+  if (!parsedElectionId.success) {
+    throw new BadRequestError(parsedElectionId.error.name);
+  }
 
   await prisma.election.update({
-    where: { id: parsedElectionId },
+    where: { id: parsedElectionId.data },
     data: {
       name: "Updated on server",
       status: ElectionStatus.ONGOING,
     },
   });
 
-  revalidatePath("/");
+  redirect(`http://localhost:3000/elections/${electionId}`);
+}
+
+async function getElections() {
+  const res = await prisma.election.findMany();
+  if (!res) {
+    throw new NotFound("");
+  }
+  return res;
+}
+
+async function findByNameOrStatus({ name, status }: FindElection) {
+  const resultElection = await getElections();
+  const result = resultElection.find(
+    (e) => e.name === name && e.status === status,
+  );
+  return result;
 }
 
 export async function addElection(formData: FormData) {
   const data = Object.fromEntries(formData);
-  const parsedData = createElectionValidator.parse(data);
+  const parsedData = createElectionValidator.safeParse(data); // Error empty input
+  if (!parsedData.success) {
+    throw new BadRequestError(parsedData.error.name);
+  }
+  const electionExist = await findByNameOrStatus(parsedData.data); // Error duplicate data
```

Comment:
Can you add to your PR samples on how the error messages are looking in case of a failed parse please? 
Thank you!

Reply:
I didn't understand what you meant by this. I modified the error message so that the BadRequestError has its own message instead of just relying on the error message from Zod. Could you provide more context about what you're asking?

Reply:
here [7a991c7](https://github.com/podcodar/voting-system/pull/67/commits/7a991c78dd714d12c15d454d73a842420dc1bf30)